sapp/sarif.py

```diff
@@ -64,19 +79,137 @@ def __init__(
 
         tool_warning_messages = get_warning_message_range(
             session,
-            self._tool_warning_code_ranges[tool][0],
-            self._tool_warning_code_ranges[tool][1],
+            self._tool_warning_code_ranges[self.tool][0],
+            self._tool_warning_code_ranges[self.tool][1],
         )
         rules_json = []
         for rule in tool_warning_messages:
             rules_json.append({"id": str(rule.code), "name": rule.message})
         driver_json["rules"] = rules_json
         self.driver = driver_json
-        self.results = [issue.to_sarif(session, tool) for issue in filtered_issues]
+        self.results = [
+            self.issue_to_sarif(session, issue) for issue in filtered_issues
+        ]
+
+    def issue_to_sarif(
```

Comment:
nitpick: should all these method start with `_` because they are not supposed to be called from outside the class?

Reply:
I am not sure. I think they may be called from outside e.g., imagine a case where some UI that builds on top of SARIF and then it can call this class to get different parts of the SARIF, but I am flexible if we want to keep this private i.e add `_`.